olive/cli/base.py

```diff
@@ -39,25 +40,145 @@ def run(self):
         raise NotImplementedError
 
 
-def get_model_name_or_path(model_name_or_path) -> Union[str, Dict[str, str]]:
-    pattern = r"^(?P<registry_name>[^:]+):(?P<model_name>[^:]+):(?P<version>[^:]+)$"
-    match = re.match(pattern, model_name_or_path)
+def _get_hf_input_model(args, model_path):
+    print("Loading HuggingFace model from:", model_path)
+    input_model = {
+        "type": "HfModel",
+        "model_path": model_path,
+        "load_kwargs": {
+            "trust_remote_code": args.trust_remote_code,
+            "attn_implementation": "eager",
+        },
+    }
+    if args.task:
+        input_model["task"] = args.task
+    return input_model
+
+
+def _get_onnx_input_model(model_path):
+    print("Loading ONNX model from:", model_path)
+    return {
+        "type": "OnnxModel",
+        "model_path": model_path,
+    }
+
+
+def _get_pt_input_model(args, model_path):
+    if not args.model_script:
+        raise ValueError("model_script is not provided. Either model_name_or_path or model_script is required.")
+
+    print("Loading PyTorch model from:", model_path)
+    user_module_loader = UserModuleLoader(args.model_script, args.script_dir)
+    input_model_config = {
+        "type": "PyTorchModel",
+        "model_script": args.model_script,
+    }
+
+    if args.script_dir:
+        input_model_config["script_dir"] = args.script_dir
+
+    # model path has high priority than model loader
+    if model_path:
+        input_model_config["model_path"] = model_path
+    else:
+        if not user_module_loader.has_function("_model_loader"):
```

Comment:
There can be cases where both model_path and model_loader are present. For example, a user has a custom model architecture and a weights file. they would need both.

Reply:
Good call! Added this case support.